src/ngraph/runtime/interpreter/int_backend.hpp

```diff
@@ -508,7 +508,7 @@ class ngraph::runtime::interpreter::INTBackend : public Backend
             {
             case element::Type_t::boolean:
                 reference::convert<T>(
-                    static_cast<const T*>(args[0]), static_cast<char*>(out[0]), element_count);
+                    static_cast<const T*>(args[0]), static_cast<bool*>(out[0]), element_count);
```

Comment:
I'm not sure this is the right way to do it, because the C++ standard does not require that `sizeof(bool) == 1`, but nGraph assumes that we are using one byte for each value in an `element::boolean` tensor. I think maybe we should have a special-case `reference::convert_to_bool` that does something like `static_cast<char>(static_cast<bool>(x))`.

(Similar comment applies to the conversion in the CPU backend, with the added awesomeness that I don't know how to effect the conversion we want through Eigen.)

Reply:
Slight clarification: _technically_ nGraph only specifies element representations in host-side buffers---internally, a backend can store a boolean in 32 bits or 27 bits or whatever it wants---_but_ the CPU and interpreter backends have chosen to have internal storage follow the host-side conventions (at least at the per-element level).